Analysis/Tasks/PWGCF/o2femtodream/include/o2femtodream/FemtoDreamContainer.h

```diff
@@ -64,30 +64,50 @@ class FemtoDreamContainer
   //  static float getkstar(const ROOT::Math::PtEtaPhiMVector& part1, const ROOT::Math::PtEtaPhiMVector& part2);
   static float getkstar(const TLorentzVector& part1, const TLorentzVector& part2);
 
- protected:
-  Observable mFemtoObs = Observable::kstar;
-
-  float mMassOne = 0.f;
-  float mMassTwo = 0.f;
+  /// constructor for kT calculation:
+  /// implemented to have as arguments some particle objects T or TLorenzVector
+ template <typename T>
+ static float getkT(const T& part1, const float mass1, const T& part2, const float mass2)
+ {
+   TLorentzVector vecpart1, vecpart2;
+   vecpart1.SetXYZM(part1.pt(), part1.eta(), part1.phi(), mass1);
+   vecpart2.SetXYZM(part2.pt(), part2.eta(), part2.phi(), mass2);
+   return getkT(vecpart1, vecpart2);
+ }
+ /// \todo improve performance of the computation by replacing TLorentzVector
+ //  static float getkT(const ROOT::Math::PtEtaPhiMVector& part1, const ROOT::Math::PtEtaPhiMVector& part2);
+ static float getkT(const TLorentzVector& part1, const TLorentzVector& part2);
+
+protected:
+ Observable mFemtoObs = Observable::kstar;
+
+ float mMassOne = 0.f;
+ float mMassTwo = 0.f;
 };
 
 void FemtoDreamContainer::init(o2::framework::HistogramRegistry* registry, Observable obs)
 {
   std::string femtoObs;
+  std::string femtokT;
   if (mFemtoObs == Observable::kstar) {
     femtoObs = "#it{k}^{*} (GeV/#it{c})";
+    femtokT = "#it{k}_{T} (GeV/#it{c})";
```

Comment:
that's not needed here, since we will always have kT as observable. So you can directly declare this at the axis